Documentation/ceph-csi-drivers.md

```diff
@@ -175,3 +175,91 @@ CSI-Addons supports the following operations:
   - [Annotating PersistentVolumeClaims](https://github.com/csi-addons/kubernetes-csi-addons/blob/v0.3.0/docs/reclaimspace.md#annotating-perstentvolumeclaims)
 - Network Fencing
   - [Creating a NetworkFence](https://github.com/csi-addons/kubernetes-csi-addons/blob/v0.3.0/docs/networkfence.md)
+
+### Enable RBD Encryption Support
+
+Ceph-CSI supports encrypting RBD volumes with LUKS encryption. More details can be found
+[here](https://github.com/ceph/ceph-csi/blob/v3.5.1/docs/deploy-rbd.md#encryption-for-rbd-volumes)
+with full list of supported encryption configurations. A sample configmap can be found
+[here](https://github.com/ceph/ceph-csi/blob/v3.5.1/examples/kms/vault/kms-config.yaml).
+
+Following steps demonstrate how to enable support for encryption:
+
+* Create `rook-ceph-csi-kms-config` configmap with required encryption configuration in
+the same namespace where Rook operator is deploy. An example is shown below:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: rook-ceph-csi-kms-config
+  namespace: rook-ceph
+data:
+  config.json: |-
+    {
+      "user-secret-metadata": {
+        "encryptionKMSType": "metadata",
+        "secretName": "storage-encryption-secret"
+      }
+    }
+```
+
+* Update the `rook-ceph-operator-config` configmap and patch the
+ following configurations
+
+```bash
+kubectl patch cm rook-ceph-operator-config -nrook-ceph -p $'data:\n "CSI_ENABLE_ENCRYPTION": "true"'
+```
+
+* Create necessary resources (secrets, configmaps etc) as required by the encryption type.
+In this case, create `storage-encryption-secret` secret in the namespace of pvc as shown:
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: storage-encryption-secret
+  namespace: rook-ceph
+stringData:
+  encryptionPassphrase: test-encryption
+```
+
+* Create a new storageclass with additional parameters `encrypted: "true"` and
+`encryptionKMSID: "<key used in configmap>"`. An example is show below:
+
+```yaml
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: rook-ceph-block-enc
+parameters:
+# additional parameters required for encryption
+  # (optional) Instruct the plugin it has to encrypt the volume
+  # By default it is disabled. Valid values are "true" or "false".
+  # A string is expected here, i.e. "true", not true.
+  encrypted: "true"
+   
+  # (optional) Use external key management system for encryption passphrases by
+  # specifying a unique ID matching KMS ConfigMap. The ID is only used for
+  # correlation to configmap entry.
+  encryptionKMSID: "user-secret-metadata"
+# ---
+
+  clusterID: rook-ceph
```

Comment:
In the yaml examples I think it works best when only the needed properties for the example are shown, and exclude properties that also apply to other examples. Then in the examples folder the full example storage class yaml is also saved where they can refer to the full example. This would mean you can simply by removing lines 248-262 from this example in the doc.